### PR TITLE
Update screenocr extension

### PR DESCRIPTION
**Pull request by xilopaint**
## Description

- Migrated codebase to use new Swift macros
- Replaced all HUDs with emojis with styled toast messages for user feedback
- Added list section to the `Select Recognition Languages` command
- Grouped checkbox preferences into a single section

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by neo773** [comment]
Could you revert the formatting changes? I guess you used prettier?

**Comment by neo773** [comment]
Could you attach a screen recording of the changes?

**Comment by xilopaint** [comment]
> Could you attach a screen recording of the changes?

What do you mean by this? If you want to know the effect of Prettier 3.3.3 in the codebase you can fork your extension, update Prettier and run `npm run fix-lint`.

File: extensions/screenocr/src/utils.ts [hunk 1]
```diff
@@ -1,54 +1,26 @@
...
-    if (preference.customWordsList) {
-      args.push("--customwordslist", preference.customWordsList);
-    }
-
-    args.push("--languages");
-    args.push(languages.map((lang) => lang.value).join(" "));
-
-    const { stdout } = await execa(command, args);
-    return stdout;
+    const recognizedText = await recognizeTextSwift(
```
Comment (neo773) [inline extensions/screenocr/src/utils.ts:10]: This is bad pattern when you have more than 2 args, use the previous code style of flags.

Comment (xilopaint) [reply extensions/screenocr/src/utils.ts:10]: Here's how you can review this PR: https://developers.raycast.com/basics/review-pullrequest

Send me how you want the code to look, but use the new macros.

**Comment by pernielsentikaer** [comment]
What's the current status of this one?

**Comment by xilopaint** [comment]
@pernielsentikaer For me, it's good. I suggested to @neo773 that he send the code with the changes he would like to make, but he hasn't sent anything so far.

**Comment by neo773** [comment]
Just tried the code, I wasn't aware of the macros that they generate type safe definition.
If you can make the args an object that'd be better.

Looks good otherwise.

**Comment by xilopaint** [comment]
@pernielsentikaer The code is finished from my part. @neo773 can build their implementation in a new PR if they find a way.

**Comment by pernielsentikaer** [comment]
> If you can make the args an object that'd be better.
Did you see this @xilopaint

**Comment by pernielsentikaer** [comment]
Let's get this PR released and then we can iterate on it

**Review by pernielsentikaer — APPROVED** [review]

**Pull request merged**